modules/gitops-configure-rollout-route-traffic-using-openshift-routes.adoc

```diff
@@ -0,0 +1,282 @@
+.. Select the *YAML* tab and view the following snippet:
++
+.Example: `Route`
+[source,yaml]
+----
+kind: Route
+metadata:
+  name: rollouts-demo-route
+spec:
+  alternateBackends:
+  - kind: Service
+    name: argo-rollouts-canary-service
+    weight: 0 #<1>
+  # (...)
+  to:
+    kind: Service
+    name: argo-rollouts-stable-service
+    weight: 100 #<2>
+----
+<1> This means that 0% of traffic is directed to the canary version.
+<2> This means that 100% of traffic is directed to the stable version.
+. Simulate the new canary version of the application by modifying the container image deployed in the rollout. 
+.. In the *Administrator* perspective of the web console, go to *Operators* -> *Installed Operators* -> *Red Hat OpenShift GitOps* -> *Rollout*.
+
+.. Select the existing *Rollout* and modify the `.spec.template.spec.containers.image` value from `argoproj/rollouts-demo:blue` to `argoproj/rollouts-demo:yellow`.
++
+As a result, the container image deployed in the rollout is modified and the rollout initiates a new canary deployment.
++
+[NOTE]
+====
+As per the `setWeight` property defined in the `.spec.strategy.canary.steps` field of the `Rollout` resource, initially 30% of traffic to the route reaches the canary version and 70% of traffic is directed towards the stable version. The rollout is paused after 30% of traffic is directed to the canary version.
+====
++
+.Example route with 30% of traffic directed to the canary version and 70% directed to the stable version.
+[source,yaml]
+----
+spec:
+  alternateBackends:
+  - kind: Service
+    name: argo-rollouts-canary-service
+    weight: 30
+  # (...)
+  to:
+    kind: Service
+    name: argo-rollouts-stable-service
+    weight: 70
+----
+
+. Simulate another new canary version of the application by running the following command in the Argo Rollouts CLI:
++
+[source,yaml]
+----
+$ oc argo rollouts promote rollouts-demo -n <namespace> <1>
+----
+<1> Specify namespace where the `Rollout` resource is defined.
++
+This increases the traffic weight to 60% in the canary version and 40% in the stable version. 
++
+.Example route with 60% of traffic directed to the canary version and 40% directed to the stable version.
+[source,yaml]
+----
+spec:
+  alternateBackends:
+  - kind: Service
+    name: argo-rollouts-canary-service
+    weight: 60
+  # (...)
+  to:
+    kind: Service
+    name: argo-rollouts-stable-service
+    weight: 40
+----
+. Increase the traffic weight in the canary version to 100% and discard the traffic in the old stable version of the application by running the following command:
++
+[source,yaml]
+----
+$ oc argo rollouts promote rollouts-demo -n <namespace> <1>
+----
+<1> Specify namespace where the `Rollout` resource is defined.
++
+.Example route with 0% of traffic directed to the canary version and 100% directed to the stable version.
+[source,yaml]
+----
+spec:
+  alternateBackends:
+  - kind: Service
+    name: argo-rollouts-canary-service
+    weight: 0
```

Comment:
In this case, I am not able to view the service `argo-rollouts-canary-service` at all in the route once `argo-rollouts-stable-service` turns 100%

Reply:
Hi @ varshab1210 - What is the reason behind this issue? Is it because of an incorrect setup or running a wrong command? In @jgwest's walkthrough, we successfully tested the entire scenario.

Reply:
I am not sure about that, @jgwest would know better.

Reply:
I am seeing the same behaviour as @varshab1210: I only see the stable service defined in Route at this step.

The logic is still correct: it's fine to remove alternateBackend when it has `weight: 0`. And I've verified that the canary weight is added back correctly when running the promotion steps.

HOWEVER, it's strange, as I don't recall us actually having logic to do that, so I will double check to determine who is making this change, and we can update the doc.

Reply:
OK, I stand corrected, [we do have logic that removes alternateBackend when weight: 0](https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-openshift/blob/8b4125a7f9ecffb0247df91a4c890f88c0c523b7/pkg/plugin/plugin.go#L138). 

@Dhruv-Soni11 you can remove this section of the YAML from the example; I can confirm this section does not appear in the Route when canary has weight=0.
```
  alternateBackends:
  - kind: Service
    name: argo-rollouts-canary-service
    weight: 0
```

Reply:
Thanks for checking that @jgwest!

Reply:
Fixed, I have removed this part of the YAML code. Thanks for clarifying this @varshab1210 @jgwest.